backend/WebUI/api_webui.go

```diff
@@ -1617,15 +1815,86 @@ func GetRegisteredUEContext(c *gin.Context) {
 		}
 	} else {
 		c.JSON(http.StatusInternalServerError, gin.H{
-			"cause": "No AMF Found",
+			"cause": "No AMF found. Is the core running?",
 		})
 	}
 }
 
-func GetUEPDUSessionInfo(c *gin.Context) {
+// not required, AMF currently returns all contexts at once
+//func GetAmfUeContextByID(c *gin.Context) {
+//	setCorsHeader(c)
+//
+//	logger.ProcLog.Infoln("Get AMF UE Context By ID")
+//
+//	webuiSelf := webui_context.GetSelf()
+//	webuiSelf.UpdateNfProfiles()
+//
+//	supi, supiExists := c.Params.Get("supi")
+//
+//	if !supiExists {
+//		logger.ProcLog.Errorln("GetAmfUeContextByID err: missing supi parameter")
+//		c.JSON(http.StatusInternalServerError, gin.H{})
+//		return
+//	}
+//
+//	// TODO: support fetching data from multiple AMFs
+//	if amfUris := webuiSelf.GetOamUris(models.NfType_AMF); amfUris != nil {
+//		requestUri := fmt.Sprintf("%s/namf-oam/v1/registered-ue-context/%s", amfUris[0], supi)
+//
+//		req, err := http.NewRequestWithContext(context.Background(), http.MethodGet, requestUri, nil)
+//		if err != nil {
+//			logger.ProcLog.Error(err)
+//			c.JSON(http.StatusInternalServerError, gin.H{})
+//			return
+//		}
+//		resp, err := httpsClient.Do(req)
+//		if err != nil {
+//			logger.ProcLog.Error(err)
+//			c.JSON(http.StatusInternalServerError, gin.H{})
+//			return
+//		}
+//		defer func() {
+//			if closeErr := resp.Body.Close(); closeErr != nil {
+//				logger.ProcLog.Error(closeErr)
+//			}
+//		}()
+//
+//		// Filter by tenant.
+//		tenantId, err := GetTenantId(c)
+//		if err != nil {
+//			logger.ProcLog.Errorln(err.Error())
+//			c.JSON(http.StatusBadRequest, gin.H{
+//				"cause": "Illegal Token",
+//			})
+//			return
+//		}
+//
+//		if tenantId == "" {
+//			sendResponseToClient(c, resp)
+//		} else {
+//			sendResponseToClientFilterTenant(c, resp, tenantId)
+//		}
```

Comment:
Do we need to remove unused part?

Reply:
I leave that to you, I don't want to refactor too much in the backend.